protected/humhub/modules/space/models/Space.php

```diff
@@ -605,7 +605,7 @@ public function getOwnerUser()
      *
      * @return array user groups
      */
-    public function getUserGroups()
+    public static function getUserGroups()
```

Comment:
We need to be careful with such changes:

![grafik](https://user-images.githubusercontent.com/9800854/101021700-89d55180-3570-11eb-84c7-566379848302.png)

Reply:
Do you mean we should replace all code like `$space->getUserGroups()` with `Space::getUserGroups()`?

Reply:
Probably it makes sense not to make the method static. That way we can support dynamic user groups later. 

What do you think? @buddh4 @yurabakhtin

Reply:
In this case we need an additional static getDefaultUserGroups for the default permissions anyway.

Reply:
I think its ok to make it static but we should align the other calls. Its possible that non static calls to static functions will be deprecated in a futre PHP version also its just not the right syntax and IDE's and code analysis tools probably will complain.

One minor concern is that a static function can not be part of a possible interface unifying ContentContainerActiveRecords (Space/user) in the future.

Reply:
I was not sure how we can convert methods to different type, because I don't know how often this method may be used in external modules, so I was asking here https://github.com/humhub/humhub/issues/4670#issuecomment-735766589 and @luke- answered "GetUserGroups: We can mark them as static.".
Of course it was my not enough clean converting in all places, so I would agree with @buddh4 because we really cannot be sure how it may be deprecated in future PHP versions.

But i left the codes without modifications because currently PHP works well with such code, and my IDE PhpStorm 2020.3 doesn't show any warnings(suggestions to some replacing) even with "PHP language level = 8.0".

It would be good if someone of both you tell me exactly I should convert all rest or keep it as is now, thanks.

Reply:
Since @buddh4 also has no problem with us marking the method as `static`, I would leave it as it is.

Also, in the examples in the PHP documentation there are also static calls to a variable: 
https://www.php.net/manual/en/language.oop5.static.php

But let's just change the few calls to Space::getUserGroups, especially where it is possible.